### PR TITLE
feat: add flag to allow disabling auto decompression by client

### LINKED ISSUE #827: Allow disabling automatic decompression when downloading files
**Is your feature request related to a problem? Please describe.**
When sending files downloaded from GCS back to the user, it would be nice if there existed an option to disable decompressing files that are compressed at rest. Currently we end up decompressing on download (well, this library does) and compressing on the way out to clients again.

 **Describe the solution you'd like**
A flag in `file.download()` that can disable `gzip` compression (I absolutely think decompressing should be the default).

### PR DESCRIPTION
**Pull request by AVaksman**
Fixes #827
- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/file.ts [hunk 1]
```diff
@@ -359,6 +359,7 @@ export interface CreateReadStreamOptions {
   validation?: 'md5' | 'crc32c' | false | true;
   start?: number;
   end?: number;
+  returnCompressed?: boolean;
```
Comment (stephenplusplus) [inline src/file.ts]: How about `decompress`, default `true`?

Comment (AVaksman) [reply src/file.ts]: done

File: src/file.ts [hunk 2]
```diff
@@ -1113,8 +1113,8 @@ class File extends ServiceObject<File> {
    *     NOTE: Byte ranges are inclusive; that is, `options.start = 0` and
    *     `options.end = 999` represent the first 1000 bytes in a file or object.
    *     NOTE: when specifying a byte range, data integrity is not available.
-   * @property {booblean} [returnCompressed] Disable auto decompression of the
-   *     received data. By default this option set to `false`.
+   * @property {booblean} [decompress] Disable auto decompression of the
```
Comment (stephenplusplus) [inline src/file.ts]: I think if you put `[decompress=true]`, that'll tip off the docs site that the default value is `true`. Also, I think it's just called a `boolean` 😅.

Comment (AVaksman) [reply src/file.ts]: Done! 😹

**Comment by JustinBeckwith** [comment]
@frankyn @jkwlui mind taking a look?

File: src/file.ts [hunk 3]
```diff
@@ -1120,6 +1121,10 @@ class File extends ServiceObject<File> {
    *     NOTE: Byte ranges are inclusive; that is, `options.start = 0` and
    *     `options.end = 999` represent the first 1000 bytes in a file or object.
    *     NOTE: when specifying a byte range, data integrity is not available.
+   * @property {boolean} [decompress=true] Disable auto decompression of the
+   *     received data. By default this option set to `true`.
```
Comment (frankyn) [inline src/file.ts]: this option *is* set to `true`

File: src/file.ts [hunk 4]
```diff
@@ -1120,6 +1121,10 @@ class File extends ServiceObject<File> {
    *     NOTE: Byte ranges are inclusive; that is, `options.start = 0` and
    *     `options.end = 999` represent the first 1000 bytes in a file or object.
    *     NOTE: when specifying a byte range, data integrity is not available.
+   * @property {boolean} [decompress=true] Disable auto decompression of the
```
Comment (frankyn) [inline src/file.ts]: Is this from the perspective of the GCS service?

File: src/file.ts (hunk 3 above)
Comment (AVaksman) [reply src/file.ts]: Currently nodejs-storage client _does_ decompress by default if data received from the server is compressed 
i. e. response contains header `content-encoding: 'gzip'`

`decompress` property seemed a better sounding name vs originally offered `returnCompressed` as per suggestion [above](https://github.com/googleapis/nodejs-storage/pull/850#discussion_r323419992)

File: src/file.ts (hunk 4 above)
Comment (AVaksman) [reply src/file.ts]: No, nodejs-sotage API always requests that the data should be served as is (compressed if applicable) i. e. sending appropriate [headers](https://github.com/googleapis/nodejs-storage/blob/b2660d8c7d68058634a20ae923afe6a44abe6b4e/src/file.ts#L1239-L1242)
```ts
const headers = {
  'Accept-Encoding': 'gzip',
  'Cache-Control': 'no-store',
} as Headers;
```

This flag is only from the perspective of nodejs-storage client to allow this library to not transcode the data if received from the server in compressed form and return to user compressed..

File: src/file.ts (hunk 3 above)
Comment (stephenplusplus) [reply src/file.ts]: There is a missing "is" in the sentence: "By default this option set to `true`."

Comment (AVaksman) [reply src/file.ts]: > There is a missing "is" in the sentence: "By default this option set to `true`."

fixed, thanks

**Comment by stephenplusplus** [comment]
@AVaksman this also fixes #709, right?

File: src/file.ts (hunk 4 above)
Comment (stephenplusplus) [reply src/file.ts]: @frankyn ping!

File: test/file.ts [hunk 5]
```diff
@@ -1203,6 +1203,17 @@ describe('File', () => {
           .resume();
       });
 
+      it('should not gunzip the response if "retunCompressed: true" is passed', done => {
```
Comment (jkwlui) [inline test/file.ts]: nit: returnCompressed => decompress

**Comment by AVaksman** [comment]
> @AVaksman this also fixes #709, right?

@stephenplusplus No, the bottom line of #709 is that server does not respect `cache-control: no transform` header from JSON API (only XML) and does not serve doubly compressed data (for example `jpg` + `gzip`), but rather automatically removes the top level compression (`gzip`) and serves the data singly compressed (just `.jpg`).

This PR cover the case when user is downloading `txt.gzip` file. Upon receiving `content-encoding: gzip`, the library would auto decompress the data and return to user (just `.txt`). Here we allowing the user to request from the library not to auto-decompress the data upon receipt but rather pass it to the user as is (`.txt.gzip`).

File: test/file.ts (hunk 5 above)
Comment (AVaksman) [reply test/file.ts]: Thanks, fixed

**Pull request merged**